### PR TITLE
Rename ID fields to align with domain terminology, and maintain views in this context

### LINKED ISSUE #120: Improve docker-compose examples to use Docker secrets for credentials passed to Hasura
Waiting for https://github.com/hasura/graphql-engine/issues/3989

### LINKED ISSUE #173: Bring Postgres view management back into this conext
The Postgres views required by this API are currently being applied directly to the `cardano-db-sync` schema, which is not ideal for maintenance, and results in all consumers of the service having a schema that's intended for another application. With Hasura we can manage the views, including migrations, so will be best to bring this back before 1.0.0

### LINKED ISSUE #174: Rename *.id to *.hash
Should be using the domain terminology here as the property name

### PR DESCRIPTION
**Pull request by rhyslbw**
This PR brings the PostrgreSQL view management back to the current context, as the tooling in Hasura has improved to better suite our workflow. It was done in this PR to make a single reverting change in `cardano-db-sync`, removing the application-specific code and re-establishing the correct dependency hiarachy.

~~@disassembler  As Hasura 1.2.1 introduces a new metadata format (`v2`), it was best to upgrade now, so the nix service will need to be upgraded, and [hasura-cli](https://hasura.io/docs/1.0/graphql/manual/hasura-cli/index.html) packaged to now [apply the migrations](https://hasura.io/docs/1.0/graphql/manual/hasura-cli/hasura_migrate_apply.html). It also makes sense to use [metadata apply](https://hasura.io/docs/1.0/graphql/manual/hasura-cli/hasura_metadata_apply.html) instead of the current HTTP API call.~~ **Deferred. Tracking in #178**

## Breaking Changes :warning: 
- Previously fields referred to as `id` are now `hash`
- `Block.merkleRootHash` is now just `merkelRoot`

Closes #173 Closes #174 Closes #120 
~~Draft until https://github.com/input-output-hk/cardano-db-sync/pull/92 is merged~~

### CODE REVIEW

**Comment by rhyslbw** [comment]
@KtorZ The diff is large due to bringing the application-specific views into the codebase, and changing the metadata format, but the primary aspect to review is the interface changes. [Viewing the .graphql file](https://github.com/input-output-hk/cardano-graphql/pull/175/files?file-filters%5B%5D=.graphql) diff shows you exactly this

**Comment by rhyslbw** [comment]
Rebuilding the Jenkins postgres-data volume now, and will rerun the tests once complete.
```
[db-sync-node:Info:60] [2020-05-14 11:29:04.64 UTC] epochPluginInsertBlock: Inserting row in epoch table for epoch 22
[db-sync-node:Info:60] [2020-05-14 11:29:13.68 UTC] insertABlock: slot 500000, block 499937, hash 226ef612830945c3068da0f2298d88aa45bf3ae19364ff1e1f88c536b8ccf503
[db-sync-node:Info:60] [2020-05-14 11:29:26.11 UTC] insertABlock: slot 505000, block 504937, hash 5f846e22ce7e1bcc6f0950b3356c375cfe93175e2b2e6da8a68b812f8f897cbe
[db-sync-node:Info:60] [2020-05-14 11:29:40.31 UTC] insertABlock: slot 510000, block 509937, hash b5dc28679903b7d97feb172809a364da9ff4b444a61d0666a29507f0b8847082
[db-sync-node:Info:60] [2020-05-14 11:29:52.54 UTC] insertABlock: slot 515000, block 514935, hash 388e22deb7e6045492912f1b451afb8fd457ba3529374e5913ffcadcbec4a88c
[db-sync-node:Info:60] [2020-05-14 11:29:59.72 UTC] insertABOBBoundary: epoch 24 hash 471ee59bee5cadc22ec85c4519acd4ee6f843eb30b34793db6ba1a9eb0426afb
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,17 +9,19 @@ Changelog
...
-- The aggregated and known very large numbers are now typed as String. Cardano JS has utilities to work with these return values, currently limited to currency conversion.
-- `cardano` query return value is accurately defined to always return an object of type `Cardano` 
-- `Transaction.fee` was a String, now a `BigInt`
 - Allow explicit ordering of `Transaction.input` and `Transaction.outputs` by their natural `index`
+- `Cardano` now matches the postgres view, and is an improvement over the previous version which performed two queries.
+
+## Breaking changes
+- Transaction and Block IDs are now labelled as `hash`, aligning with the domain terminology.
+- `Block.merkelRootHash` -> `Block.merkelRoot`
+- The aggregated and known very large numbers are now typed as String. Cardano JS has utilities to work with these return values, currently limited to currency conversion.
```
Comment (KtorZ) [inline CHANGELOG.md:18]: I still find this one weird. Likely, for numbers that are this big (> 2^53), we probably don't need an enormous precision, so it'd be fine to store them as two numbers with an exponent and a coefficient :thinking: ?

This is really just a thought.

Comment (rhyslbw) [reply CHANGELOG.md:18]: I agree, however with the current design the `cardano-graphql` server is just routing requests and responses to and from Hasura to keep the complexity low. I'd like to revisit this before the next major release

**Comment by rhyslbw** [comment]
There currently a conflict with the migration strategy in `cardano-db-sync`, where it drops and re-applies views on each startup, whereas Hasura tracks the catalog version to make a decision. I will have to modify our start procedure and create a custom Docker image that uses a similar strategy of rolling back to the first version, then forward to the latest. The custom Docker image was already on the cards in https://github.com/input-output-hk/cardano-graphql/issues/120

**Pull request merged**